### DISCUSSION TITLE
How to run Introspect from nodejs?

### DISCUSSION THREAD

**Discussion by mariusa** [prisma/prisma#5074] [Q&A]
Hi, I'm looking to run Introspect from nodejs, similar to sequelize-auto (https://github.com/sequelize/sequelize-auto)

```js
let SequelizeAuto = require('sequelize-auto')
    let sqauto = new SequelizeAuto(db, null, null, {
        //directory: false, // false: prevents the program from writing to disk
        //camelCase: true, // convert snake_case column names to camelCase field names: user_id -> userId
        camelCaseFileName: false, // file names created for each model use camelCase.js not snake_case.js
        additional: {
            timestamps: false
        },
        views: true
      
    })
    let sa = await sqauto.run() //this generates models
```

This should be possible with prismal too, eg https://github.com/prisma/prisma/blob/7a9ac05d30903621d070df1d58137ce50c3523af/src/packages/cli/src/__tests__/introspect.test.ts

but I can't figure out how exactly that works. Would you kindly provide a similar script to above?
Thanks

**Comment by ryands17** [comment]
Hey @mariusa 👋 

Currently there's not programmatic API to run Introspections so as a workaround you would need to do something like this:

```js
const util = require('util')
const exec = util.promisify(require('child_process').exec)

await exec(`yarn prisma introspect`) // or await exec(`npx prisma introspect`)
```

Let me know if this works :)

**Reply by yardenshoham** [reply]
When will there be a programmatic API for prisma? When I have a fresh env without npx/yarn, I'd want my app to initialize the database...
Context: https://github.com/prisma/prisma/issues/4703#issuecomment-761212458

**Reply by ryands17** [reply]
@yardenshoham 👋 

It's not currently on our [Roadmap](https://www.notion.so/prismaio/Prisma-public-roadmap-50766227b779464ab98899accb98295f) so I cannot provide an exact timeline.
But do add a 👍 to this request so that we can look into the priority for this and follow the request for more updates :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
